### PR TITLE
Update deployment for cluster autoscaler support

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/2203

### CODE REVIEW

File: chart/questions.yaml [hunk 1]
```diff
@@ -497,6 +497,17 @@ WARNING:
     min: 0
     max: 40
     default: 12
+- variable: defaultSettings.kubernetesClusterAutoscalerEnabled
+    label: Kubernetes Cluster Autoscaler Enabled (Experimental)
+    description: "Enable this setting will notify Longhorn that the cluster is using Kubernetes cluster-autoscaler.
```
Comment (innobead) [inline chart/questions.yaml:502]:
```suggestion
    description: "Enabling this setting will notify Longhorn that the cluster is using Kubernetes cluster-autoscaler.
```

File: chart/questions.yaml [hunk 2]
```diff
@@ -497,6 +497,17 @@ WARNING:
...
+    Longhorn prevents data loss by only allowing CA to scale down a node that met all conditions:
```
Comment (innobead) [inline chart/questions.yaml:503]:
```suggestion
    Longhorn prevents data loss by only allowing the Cluster Autoscaler to scale down a node that met all conditions:
```

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just some rewords needed.

File: chart/questions.yaml (hunk 1 above)
Comment (c3y1huang) [reply chart/questions.yaml]: Updated.

File: chart/questions.yaml [hunk 3]
```diff
@@ -497,6 +497,17 @@ WARNING:
     min: 0
     max: 40
     default: 12
+- variable: defaultSettings.kubernetesClusterAutoscalerEnabled
+    label: Kubernetes Cluster Autoscaler Enabled (Experimental)
+    description: "Enabling this setting will notify Longhorn that the cluster is using Kubernetes Cluster Autoscaler.
+    Longhorn prevents data loss by only allowing the Cluster Autoscaler to scale down a node that met all conditions:
+      - No volume attached
```
Comment (shuo-wu) [inline chart/questions.yaml:504]: NIT: Just realized that this statement may be confusing. Should be `No volume attached to the node` instead.

File: chart/questions.yaml [hunk 4]
```diff
@@ -497,6 +497,17 @@ WARNING:
...
+      - Is not the last node containing the replica of a detached volume.
```
Comment (shuo-wu) [inline chart/questions.yaml:505]: NIT: For a running volume on another node, we cannot remove the node either when it contains the last replica.

File: chart/templates/default-setting.yaml [hunk 5]
```diff
@@ -42,3 +42,4 @@ data:
     backing-image-recovery-wait-interval: {{ .Values.defaultSettings.backingImageRecoveryWaitInterval }}
     guaranteed-engine-manager-cpu: {{ .Values.defaultSettings.guaranteedEngineManagerCPU }}
     guaranteed-replica-manager-cpu: {{ .Values.defaultSettings.guaranteedReplicaManagerCPU }}
+    kubernetes-cluster-autoscaler-enabled: {{ .Values.defaultSettings.kubernetesClusterAutoscalerEnabled }}
```
Comment (shuo-wu) [inline chart/templates/default-setting.yaml:45]: Note: We will switch to using CRs for default settings in v1.3.0. But we can upgrade this whole YAML later.

File: chart/questions.yaml (hunk 4 above)
Comment (c3y1huang) [reply chart/questions.yaml]: Fixed.

Also in https://github.com/longhorn/longhorn-manager/pull/1305

File: chart/templates/default-setting.yaml (hunk 5 above)
Comment (c3y1huang) [reply chart/templates/default-setting.yaml:45]: Fixed.

File: chart/questions.yaml (hunk 3 above)
Comment (c3y1huang) [reply chart/questions.yaml]: Fixed.

**Review by innobead — APPROVED** [review]

**Pull request merged**